### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by davidcavazos**
Initial commit for the Beam Kotlin starter project.

R: @aaltay can you help us find reviewers for Kotlin? Thanks!

### CODE REVIEW

**Comment by aaltay** [comment]
I do not know who would be a good kotlin reviewer. Could you email the team? (Or reach out to @kennknowles )

I can rubber stamp this if (i) you verified it AND (ii) we cannot find any Kotlin reviewers. Let me know.

**Comment by brucearctor** [comment]
Potentially check for the authors of:  https://github.com/apache/beam/tree/master/learning/katas/kotlin

**Comment by brucearctor** [comment]
and, @tosun-si was talking much about kotlin at BeamSummit so he might be able to review.

File: app/src/test/kotlin/beam/starter/AppTest.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+package beam.starter
+
+import org.apache.beam.sdk.testing.PAssert
+import org.apache.beam.sdk.testing.TestPipeline
+import org.junit.Rule
```
Comment (kennknowles) [inline app/src/test/kotlin/beam/starter/AppTest.kt:13]: Is junit idiomatic for Kotlin?

Comment (davidcavazos) [reply app/src/test/kotlin/beam/starter/AppTest.kt:13]: If we want to use `TestPipeline` it has to be a `@Rule` which is only supported on JUnit 4 (JUnit 5 is not supported). I guess we could use a "normal" pipeline and use Kotlin's tests.

**Comment by tosun-si** [comment]
Ok I am going to review the code tomorrow 👍🏻

File: app/src/test/kotlin/beam/starter/AppTest.kt (hunk 1 above)
Comment (tosun-si) [reply app/src/test/kotlin/beam/starter/AppTest.kt:13]: For my example presented in Beam Summit, I also used `TestPipeline` with a Junit 4 `@Rule` :

https://github.com/tosun-si/teams-league-kotlin-dlq-asgarde-beam-summit/blob/main/src/test/kotlin/fr/groupbees/domain_ptransform/TeamStatsTransformTest.kt

**Comment by davidcavazos** [comment]
Hi @tosun-si, I made the test pipeline transient like in your example. Other than that, please let me know if there are any other changes you recommend. Do the Gradle build files look good?

**Comment by tosun-si** [comment]
Hi @davidcavazos, thanks.

I have a little question, why Gradle was chosen instead of Maven for this starter. Gradle is good but I more thought about Maven to be homogeneous with the Beam Java starter. What do you think about that ?

If you prefer to keep Gradle, it's ok for me too :)

**Comment by brucearctor** [comment]
Beam https://github.com/apache/beam relies on Gradle

**Comment by tosun-si** [comment]
@brucearctor OK I understand, I was wrong in my analysis, only the Maven Archetype generates a project with a Maven pom.xml file : 

```shell
mvn archetype:generate \
    -DarchetypeGroupId=org.apache.beam \
    -DarchetypeArtifactId=beam-sdks-java-maven-archetypes-examples \
    -DarchetypeVersion=2.40.0 \
    -DgroupId=org.example \
    -DartifactId=word-count-beam \
    -Dversion="0.1" \
    -Dpackage=org.apache.beam.examples \
    -DinteractiveMode=false
```

In this case, Gradle is the good choice :)

**Comment by brucearctor** [comment]
for most, they should be interchangeable :-) 

Specifically we are considering you ( @tosun-si ) to be the Kotlin expert on this thread, and this is a Kotlin example.  Anything else stand out as suggested/needing to change?

**Comment by tosun-si** [comment]
Everything is good for me. 

Maybe in the future we can add more examples, but for a first version it's ok and very good.
Thanks @brucearctor

**Review by tosun-si — APPROVED** [review]

**Pull request merged**